bokeh/models/tools.py

```diff
@@ -413,3 +413,13 @@ class HoverTool(Tool):
         the ``Rect`` and ``Quad`` glyphs are supported.
 
     """)
+
+    mode = Enum("point", "hline", "vline", help="""
+    Whether to consider hover pointer as point (x/y values), or a
+    span on h or v directions.
+    """)
+
+    hit_value_mode = Enum("snap_to_data", "glyph_center", "hit_interpolate",
```

Comment:
I don't think we should worry **at all** about interpolating values for display here. That needs to be handled elsewhere, where the output format is specified, and asked for explicitly. This should *only* be about positioning the tooltip, conflating anything else to do with display policy would be extremely confusing.